ntbs-integration-tests/NotificationPages/ManualTestResultEditPages.cs

```diff
@@ -0,0 +1,167 @@
+            {
+                ["FormattedTestDate.Day"] = "2",
+                ["FormattedTestDate.Month"] = "2",
+                ["FormattedTestDate.Year"] = "2012",
+                ["TestResultForEdit.ManualTestTypeId"] = ((int)ManualTestTypeId.Smear).ToString(),
+                ["TestResultForEdit.SampleTypeId"] = ((int)SampleTypeId.BronchialWashings).ToString(),
+                ["TestResultForEdit.Result"] = ((int)Result.Negative).ToString(),
+            };
+            var result = await SendPostFormWithData(editDocument, formData, editUrl);
+
+            // Assert
+            result.AssertRedirectTo(GetPathForId(NotificationSubPaths.EditTestResults, NotificationId));
+            var testsListPage = await Client.GetAsync(GetRedirectLocation(result)); // Follow the redirect to see results table
+            var testsListDocument = await GetDocumentAsync(testsListPage);
+            var manualResults = testsListDocument.GetElementById("manual-results");
+
+            Assert.Contains("Smear", manualResults.TextContent);
+            Assert.Contains("Bronchial washings", manualResults.TextContent);
+            Assert.Contains("Negative", manualResults.TextContent);
+            Assert.DoesNotContain("Lung bronchial tree tissue", manualResults.TextContent);
+        }
+
+        [Fact]
+        public async Task PostEditOfManualTestResult_ReturnsAllRequiredValidationErrors()
+        {
+            // Arrange
+            var NotificationId = Utilities.DRAFT_ID;
+            var editUrl = GetCurrentPathForId(NotificationId) + TEST_ID;
+            var editPage = await Client.GetAsync(editUrl);
+            var editDocument = await GetDocumentAsync(editPage);
+
+            // Act
+            var formData = new Dictionary<string, string>
+            {
+                ["FormattedTestDate.Day"] = "",
+                ["FormattedTestDate.Month"] = "",
+                ["FormattedTestDate.Year"] = "",
+                ["TestResultForEdit.ManualTestTypeId"] = "",
+                ["TestResultForEdit.SampleTypeId"] = "",
+                ["TestResultForEdit.Result"] = "",
+            };
+            var result = await SendPostFormWithData(editDocument, formData, editUrl);
+            var resultDocument = await GetDocumentAsync(result);
+
+            // Assert
+            result.AssertValidationErrorResponse();
+
+            resultDocument.AssertErrorMessage("test-date", string.Format(ValidationMessages.RequiredEnter, "Test date"));
+            resultDocument.AssertErrorMessage("test-type", string.Format(ValidationMessages.RequiredSelect, "Test type"));
+            resultDocument.AssertErrorMessage("sample-type", string.Format(ValidationMessages.RequiredSelect, "Sample type"));
+            resultDocument.AssertErrorMessage("result", string.Format(ValidationMessages.RequiredSelect, "Result"));
+        }
+
+        [Fact]
+        public async Task PostEditWithInvalidResult_ReturnsErrors()
+        {
```

Comment:
Could we rename this to "PostEditWithInvalidTestAndSampleTypeCombination_ReturnsErrors"? Might as well be more descriptive :) Might already not be the case in other tests (where I'm pretty sure I had fairly generic names) but looking through it as part of the review I feel like I'd prefer more precise naming